lib/lotus/routing/endpoint_resolver.rb

```diff
@@ -197,7 +198,7 @@ def classify(string)
       private
       def resolve_callable(callable)
         if callable.respond_to?(:call)
-          @endpoint_class.new(callable)
+          create_endpoint(callable, @endpoint_class)
```

Comment:
@jodosha  do you think create_endpoint here is necessary? because always inside create_endpoint is checking again the respond_to and it always goes through `MiddlewareEndpoint`